sdk/identity/Azure.Identity/src/ClientCertificateCredential.cs

```diff
@@ -138,7 +129,7 @@ internal ClientCertificateCredential(string tenantId, string clientId, IX509Cert
 
             ClientId = clientId ?? throw new ArgumentNullException(nameof(clientId));
 
-            ClientCertificateProvider = certificateProvider;
+            ClientCertificateProvider = certificateProvider ?? throw new ArgumentNullException(nameof(certificateProvider));
```

Comment:
I have mixed feelings about this check since this isn't publicly callable code.

Reply:
I agree that it isn't really needed. Removed it.